### DISCUSSION TITLE
How to update Mermaid

### DISCUSSION THREAD

**Discussion by jkc20** [zauberzeug/nicegui#2652] [Q&A] [answered]
### Question

Hi,
I like to use Mermaid block diagrams introduced in v10.8.0 in ui.markdown/ui.mermaid (currently v10.5.1). I tried to change the files under ../elements/lib/mermaid without success. How could I achieve this in my local installation ?
Thanks in advance for any help.

**Comment by rodja** [comment, marked as answer]
Have a look at `npm.py` in the root dir of this repo. It's run to update the dependencies according to the npm.json config.

**Comment by stephprobst** [comment]
I know this is old, but just in case someone else comes looking for a simple solution (as I did). I'm patching mermaid on application start like this:

```python
import shutil
from pathlib import Path
import nicegui
nicegui_path = Path(nicegui.__file__).parent

# Original mermaid file path
original_file = nicegui_path / "elements" / "lib" / "mermaid" / "mermaid.esm.min.mjs"

# Custom mermaid file path
current_dir = Path(__file__).parent
custom_file = current_dir / "mermaid.esm.min.mjs"

# Replace the original file with custom version
shutil.copy2(custom_file, original_file)
```

For this to work, the custom mermaid file needs to be bundled into a single file like so:

```
esbuild node_modules/mermaid/dist/mermaid.esm.min.mjs --bundle --format=esm --outfile=mermaid.esm.min.mjs
```

Not the cleanest approach, but it works for me...

**Comment by evnchn** [comment]
This may work too: https://github.com/zauberzeug/nicegui/discussions/4739#discussioncomment-13141055
